synapse/storage/prepare_database.py

```diff
@@ -619,9 +619,10 @@ def _get_or_create_schema_state(
 
     txn.execute("SELECT version, upgraded FROM schema_version")
     row = txn.fetchone()
-    current_version = int(row[0]) if row else None
+    current_version = int(row[0]) if row is not None and len(row) > 0 else None
 
     if current_version:
+        assert row is not None
```

Comment:
(needed for mypy to stop talking about `row` being `Optional[]`, is already assured by `current_version` being truthy)

Reply:
It's not due to the change above, but `txn.fetchone()` is `Optional[]`, and while mypy does not do proper advanced branch checking to see that `current_version` is truthy if `row` is truthy (or `not None`), to see that correlation.

The above change (`row` being truthy vs `row` being `not None`) didn't change this error, i put this assert here to assure mypy that this is okay.

Reply:
So I think this code can be simplified a bit, by changing the `if current_version` to `if row is not None` and then setting `current_version = int(row[0])` as the first line in the if-statement.